keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/SortingStrategy.kt

```diff
@@ -5,8 +5,8 @@ import com.netflix.spinnaker.keel.api.schema.Discriminator
 /**
  * Strategy for how to sort versions of artifacts.
  */
-interface VersioningStrategy {
+interface SortingStrategy {
   @Discriminator
   val type: String
-  val comparator: Comparator<String>
+  val comparator: Comparator<PublishedArtifact>
```

Comment:
This is what will allow us to sort artifact "versions" based on arbitrary fields (such as branch and creation timestamp).

Reply:
That seems like a great change